extensions/Worker.Extensions.CosmosDB/src/CosmosDBConverter.cs

```diff
@@ -0,0 +1,207 @@
+
+            if (container is null)
+            {
+                throw new InvalidOperationException($"Unable to create Cosmos container client for '{cosmosAttribute.ContainerName}'.");
+            }
+
+            return await (Task<object>)createPOCOMethod.Invoke(this, new object[] { container, cosmosAttribute });
+        }
+
+        private async Task<object> CreatePOCOAsync<T>(Container container, CosmosDBInputAttribute cosmosAttribute)
+        {
+            if (String.IsNullOrEmpty(cosmosAttribute.Id) || String.IsNullOrEmpty(cosmosAttribute.PartitionKey))
+            {
+                throw new InvalidOperationException("The 'Id' and 'PartitionKey' properties of a CosmosDB single-item input binding cannot be null or empty.");
+            }
+
+            ItemResponse<T> item = await container.ReadItemAsync<T>(cosmosAttribute.Id, new PartitionKey(cosmosAttribute.PartitionKey));
+
+            if (item is null || item?.StatusCode is not System.Net.HttpStatusCode.OK || item.Resource is null)
+            {
+                throw new InvalidOperationException($"Unable to retrieve document with ID '{cosmosAttribute.Id}' and PartitionKey '{cosmosAttribute.PartitionKey}'");
+            }
+
+            return item.Resource;
+        }
+
+        private async Task<object> CreatePOCOCollectionAsync<T>(Container container, CosmosDBInputAttribute cosmosAttribute)
+        {
+            QueryDefinition queryDefinition = null!;
+            if (!String.IsNullOrEmpty(cosmosAttribute.SqlQuery))
+            {
+                queryDefinition = new QueryDefinition(cosmosAttribute.SqlQuery);
+                if (cosmosAttribute.SqlQueryParameters?.Count() > 0)
+                {
+                    foreach (var parameter in cosmosAttribute.SqlQueryParameters)
+                    {
+                        queryDefinition.WithParameter(parameter.Key, parameter.Value.ToString());
+                    }
+                }
+            }
+
+            PartitionKey partitionKey = String.IsNullOrEmpty(cosmosAttribute.PartitionKey)
+                                        ? PartitionKey.None
+                                        : new PartitionKey(cosmosAttribute.PartitionKey);
+
+            // Workaround until bug in Cosmos SDK is fixed
+            // Currently pending release: https://github.com/Azure/azure-cosmos-dotnet-v3/commit/d6e04a92f8778565eb1d1452738d37c7faf3c47a
+            QueryRequestOptions queryRequestOptions = new();
+            if (partitionKey != PartitionKey.None)
+            {
+                queryRequestOptions = new() { PartitionKey = partitionKey };
+            }
+
+            using (var iterator = container.GetItemQueryIterator<T>(queryDefinition: queryDefinition, requestOptions: queryRequestOptions))
+            {
+                if (iterator is null)
+                {
+                    var queryInfo = queryDefinition is not null ? $" SqlQuery: {queryDefinition.QueryText}." : "";
```

Comment:
Is it safe to put their query text directly into an exception? What if it has sensitive data?